samples/solutions/stingy_connection.py

```diff
@@ -0,0 +1,275 @@
+            self.log_info_and_print("Connected connected_event is set...")
+        else:
+            self.log_info_and_print("Disconnected connected_event is set...")
+
+    def is_retryable(self, exc):
+        if type(exc) in self.transient_errors:
+            return True
+        else:
+            return False
+
+    def log_error_and_print(self, s):
+        logger.error(s)
+        print(s)
+
+    def log_info_and_print(self, s):
+        logger.info(s)
+        print(s)
+
+    async def enqueue_message(self):
+        message_id = 0
+        while True:
+            msg = Message("current wind speed ")
+            msg.message_id = message_id
+            msg.content_type = "application/json"
+            self.log_info_and_print("Created a message...")
+            self.message_queue.put_nowait(msg)
+            self.message_list[message_id] = msg
+            # time between 10 seconds and less than 30 minutes.
+            randint = random.randint(UPPER_LIMIT_OF_ENQUEUEING, UPPER_LIMIT_OF_ENQUEUEING)
+            self.log_info_and_print(
+                "Will sleep for {} seconds and then enqueue messages".format(randint)
+            )
+            await asyncio.sleep(randint)
+            message_id += 1
+
+    async def do_all_tasks_and_disconnect(self):
+        self.log_info_and_print("Current qsize is: {}".format(self.message_queue.qsize()))
+        while not self.message_queue.empty():
+            msg = await self.message_queue.get()
+            task = asyncio.create_task(self.device_client.send_message(msg))
+            self.task_list.append(task)
+            self.message_queue.task_done()
+        try:
+            if self.device_client.connected:
+                self.log_info_and_print("sending messages...")
+                await asyncio.gather(*self.task_list)
+                self.log_info_and_print("sent all messages...")
+                self.task_list = []
+                await self.device_client.disconnect()
+        except Exception as e:
+            self.log_error_and_print(
+                "Caught exception while trying to send message or disconnect: {}".format(
+                    get_type_name(e)
+                )
+            )
+
+    async def connect_with_retry_send_all_and_disconnect(self):
+        while True:
+            for i in range(1, 10):
+                if not self.device_client.connected:
+                    try:
+                        self.log_info_and_print(
+                            "Attempting to connect the device client try number {}....".format(i)
+                        )
+                        await self.device_client.connect()
+                        self.log_info_and_print("Successfully connected the device client...")
+                        await self.do_all_tasks_and_disconnect()
+                        break
+                    except Exception as e:
+                        self.log_error_and_print("Caught exception while trying to connect...")
```

Comment:
>elf.log_error_and_print("Caught exception while trying to connect...") [](http://example.com/codeflow?start=25&length=70)

I recommend printing the type of exception.  Otherwise the log might not tell you what actually happened.